### ISSUE TITLE
vkAcquireNextImageKHR unclear relationship between return codes and image acquisition

### ISSUE THREAD

**Issue by Hawk777** [KhronosGroup/Vulkan-Docs#1776]
In the documentation for `vkAcquireNextImageKHR`, four return codes (`VK_SUCCESS`, `VK_TIMEOUT`, `VK_NOT_READY`, and `VK_SUBOPTIMAL_KHR`) are listed as “success”es, with a bunch of others being listed as “failure”s.

However, the immediately following paragraph states:
> When successful, `vkAcquireNextImageKHR` acquires a presentable image from swapchain that an application can use, and sets `pImageIndex` to the index of that image within the swapchain.

Intuitively, this seems clearly wrong: an image should be acquired if the result is `VK_SUCCESS` or `VK_SUBOPTIMAL_KHR`, but *not* if the result is `VK_TIMEOUT` or `VK_NOT_READY`, even though those are considered “success” results.

The next paragraph states:
> If `timeout` is zero, then `vkAcquireNextImageKHR` does not wait, and will either successfully acquire an image, or fail and return `VK_NOT_READY` if no image is available.

This seems poorly worded and potentially contradictory: if no image is immediately available, then `vkAcquireNextImageKHR` will *fail* and return… a success code?

**Comment by stonesthrow** [comment]
You are quite right of course and thank you for raising issue. I will fix the text that SUCCESS and SUBOPTIMAL will be successful with Image, the other two are failures with no image. 

I'll try to get corrected text ASAP.

**Comment by Hawk777** [comment]
There might be a bigger problem than just `vkAcquireNextImageKHR`, actually. [Section 3.8](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/chap3.html#fundamentals-returncodes) has this to say:
> Successful completion codes are returned when a command needs to communicate success or status information. All successful completion codes are non-negative values.
>
> Run time error codes are returned when a command needs to communicate a failure that could only be detected at runtime. All runtime error codes are negative values.

But `VK_TIMEOUT` and `VK_NOT_READY` are positive numbers, and that obviously can’t be changed. So changing the writeup of just `vkAcquireNextImageKHR` would then make it inconsistent with the overall spec-wide definition of success and failure.

**Comment by stonesthrow** [comment]
A very good point, thank you. I might have to show this to the main Vulkan group for comment - try to avoid that inconsistency.

**Comment by oddhack** [comment]
This should be fixed in the 1.3.239 spec update.

**Comment by Hawk777** [comment]
Hm, it looks like only the first part was fixed. I’m looking at [1.3.240](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#vkAcquireNextImageKHR), and it still says “If timeout is zero, … fail and return VK_NOT_READY if no image is available” despite VK_NOT_READY being a success code not a failure code.

**Comment by oddhack** [comment]
@stonesthrow reopening per previous comment - should go back on the SI TSG agenda?

**Comment by stonesthrow** [comment]
Yes, keep open. There are codes that are not failures, like TIMEOUT and NOT_READY, that I'm not sure the latest update covered.

**Comment by stonesthrow** [comment]
Reopen for me to check language for non-errors

**Comment by stonesthrow** [comment]
Reassign to @cubanismo to reassign, follow up text still needs update.
